tests/org/test_search_org_api.py

```diff
@@ -0,0 +1,60 @@
+from org.models import *
+from tests.AuthAPITestCase import AuthAPITestCase
+from rest_framework.test import APITestCase,APIClient
+from rest_framework import status
+from org.serializers import *
+
+class SearchOrgTestCase(AuthAPITestCase):
+    """
+    Testing the Search Org API
+    """
+    search_org_api='/api/org/'
+
+    def setUp(self):
+        """
+            Create an organization in the test database
+        """
+        #Inheriting the base class functionality
+        super(SearchOrgTestCase,self).setUp()
+        # Create the org using serializer
+        data_org = {
+                "name":'Test_Org',
+                "tagline":'testing is everything'
+        }
+        serializer = CreateOrgSerializer(data = data_org)
+        if serializer.is_valid():
+            self.org = serializer.save()[0]
+
+    def test_fail_without_auth_header(self):
+        un_auth_client = APIClient()
+        response = un_auth_client.get(self.search_org_api)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_success_with_auth_header(self):
+        auth_client = self.create_auth_client()
+        response = auth_client.get(self.search_org_api)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_success_with_all_orgs(self):
+        """
+        Testing if all the organizations are listed when 
+        nothing is specified in search param.
+        """
+        auth_client = self.create_auth_client()
+        all_orgs = ListOrgSerializer(Org.objects.all(), many=True)
+        response = auth_client.get(self.search_org_api)
+        self.assertEqual(response.data, all_orgs.data)
+
+    def test_success_search_result(self):
+        """
+        Testing if only the organisation wanted ends
+        up as a search result.
+        """
+        search_org_api = '/api/org/?search=test'
+        auth_client = self.create_auth_client()
+        filter_result = ListOrgSerializer(Org.objects.filter(name='Test_Org'), many=True)        
```

Comment:
What I interpret is that the API returns a list of orgs prefixed with text in the search query, and so I think filter should be replaced as follows.
```suggestion
        filter_result = ListOrgSerializer(Org.objects.filter(name__startswith='test'), many=True)      
```
P.S. - The test fails if I add another Org with prefix test. Tried it in my local and got an Assertion Error. 
`AssertionError: [Orde[128 chars]ne)]), OrderedDict([('id', 2), ('name', 'Test_[98 chars]e)])] != [Orde[128 chars]ne)])]`

Reply:
@anshsrtv what @priyanshisharma mentioned is the intended behavior, please make changes accordingly :innocent: